tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -836,17 +835,125 @@ export class WebGPUBackend extends KernelBackend {
+    if (useImport) {
+      bindGroupLayoutEntries.push({
+        binding: 1,
+        visibility: GPUShaderStage.COMPUTE,
+        externalTexture: {},
+      });
+    } else {
+      bindGroupLayoutEntries.push(
+          {binding: 1, visibility: GPUShaderStage.COMPUTE, texture: {}});
+    }
+    // Uniform buffer binding layout.
+    bindGroupLayoutEntries.push(
+        {binding: 2, visibility: GPUShaderStage.COMPUTE, buffer: {}});
+    const fromPixelBindGroupLayout =
+        this.device.createBindGroupLayout({entries: bindGroupLayoutEntries});
+    const fromPixelPipelineLayout = this.device.createPipelineLayout(
+        {bindGroupLayouts: [fromPixelBindGroupLayout]});
+    return {
+      bindGroupLayout: fromPixelBindGroupLayout,
+      pipelineLayout: fromPixelPipelineLayout
+    };
+  }
+
+  private copyExternalImageToTexture(
+      externalImage: ExternalImage|HTMLVideoElement,
+      outShape: number[]): GPUTextureView {
+    const textureUsage = GPUTextureUsage.COPY_DST |
+        GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.TEXTURE_BINDING;
+    const textureFormat = 'rgba8unorm' as GPUTextureFormat;
+    const texture = this.textureManager.acquireTexture(
+        outShape[1], outShape[0], textureFormat, textureUsage);
+    const externalResource = texture.createView();
+
+    this.queue.copyExternalImageToTexture(
+        {source: externalImage as ExternalImage, origin: {x: 0, y: 0}},
+        {texture}, [outShape[1], outShape[0]]);
+
+    const textureInfo = {
+      width: outShape[1],
+      height: outShape[0],
+      format: textureFormat,
+      usage: textureUsage,
+      texture
+    };
+    this.textureDisposalQueue.push(textureInfo);
+    return externalResource;
+  }
+
   runFromPixelsProgram(
-      program: FromPixelsProgram, output: GPUBuffer, layout: WebGPULayout,
-      externalResource: GPUExternalTexture|GPUTextureView, outputId: DataId) {
+      program: webgpu_program.WebGPUProgram, outShape: number[],
+      programUniforms: ProgramUniform, useImport: boolean,
+      externalImage: ExternalImage|HTMLVideoElement) {
     program.dispatch = reshapeDispatch(this.device, program);
+
+    const output = this.makeTensorInfo(outShape, 'int32');
+    if (util.sizeFromShape(output.shape) === 0) {
+      // Short-circuit the computation since the result is empty (has 0 in its
+      // shape).
+      const outData = this.tensorMap.get(output.dataId);
+      outData.values =
+          util.getTypedArrayFromDType(output.dtype as 'float32', 0);
+      return output;
+    }
+    this.uploadToGPU(output.dataId);
+    const key = webgpu_program.makeShaderKey(program, [output.shape]);
+
+    const layout = this.getFromPixelTextureLayout(useImport);
+
+    const pipeline = this.getAndSavePipeline(key, () => {
+      return webgpu_program.compileProgram(
+          this.device, program, layout.pipelineLayout, [], output, true);
+    });
+
+    let externalResource: GPUExternalTexture|GPUTextureView;
```

Comment:
(Heads up here). 
GPUExternalTexture will have an attribute named isExpire to see whether a new importExternalTexture is required. A future PR might needed to cache this object and the source video for performance too.